### ISSUE TITLE
Serializing old archives after updating the library

### ISSUE THREAD

**Issue by gokhansolak** [stulp/dmpbbo#54]
I have been using older versions of DMPBBO (~2018). Recently I updated to the latest and observed that I can't save or load serialized DMPs anymore. After some debugging, I noticed that adding the ```#include "dmp/serialization.hpp"``` line solves the saving issue. It is hard to spot since the program still builds without this header and it used to work without the header in the older versions. I suggest mentioning this in the documentation.

I still can't load the old DMP archives with the following error:
```
terminate called after throwing an instance of 'boost::archive::archive_exception'
what():  unregistered class
```
Probably, I will need to do some changes in the old XML files. I will send the updates.

**Comment by stulp** [comment]
Thanks for pointing this out. It is good to know that someone is using the serialization functionality. To be honest, getting boost:serialization to work was a real pain, and so is keeping it running.



https://github.com/stulp/dmpbbo/blob/master/README.md#why-python-and-c

**Comment by gokhansolak** [comment]
Thank you for your answer. I don't serialize bbo modules.

I actually solved the problem. I run diff between the old and new xml files and noticed that the new one doesn't contain DmpBbo namespace in class_name attributes. When I added the namespaces the boost::serialization worked. I also created a [Gist file](https://gist.github.com/gokhansolak/975c80cba7ed46419145582306fd90e5) that adds ```DmpBbo:: ``` prefix to all class names in xml files in a directory, for those who may have a similar problem.

**Comment by stulp** [comment]
Thank you for sharing your solution. Good to know that it is easy to make the XML files conform to the new syntax. I've included it in the documentation in d2fe4cb1e10f

I think this change of serialization behaviour must have happened in commit fb936a2e1ac, when I moved the registering of classes from the hpp file where the class is declared into one hpp file per module. This is less elegant and less modular, but since then, serialization works more robustly, i.e. not causing as many run-time crashes as you have experienced.
